.github/actions/docker-build-preview/action.yml

```
@@ -0,0 +1,73 @@
+name: Docker Build Preview
+description: Build & push preview Docker image from packaged ZIPs
+inputs:
+  version:
+    description: Version used for tagging and ZIP file paths
+    required: true
+  registry:
+    description: Docker registry URL
+    required: true
+  docker-username:
+    description: Docker registry username
+    required: true
+  docker-password:
+    description: Docker registry password
+    required: true
+  repository:
+    description: Docker repository name
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Prepare Docker build context
+      shell: bash
+      run: |
+        set -euo pipefail
+        VERSION="${{ inputs.version }}"
+        echo "Preparing Docker context with version: ${VERSION}"
+
+        # Core package (required)
+        if [[ -f "nuxeo-ai-core-package/target/nuxeo-ai-core-${VERSION}.zip" ]]; then
+          cp "nuxeo-ai-core-package/target/nuxeo-ai-core-${VERSION}.zip" docker/
+        else
+          echo "ERROR: Core package not found: nuxeo-ai-core-package/target/nuxeo-ai-core-${VERSION}.zip"
+          exit 1
+        fi
+
+        # Optional addon packages (copy if present)
+        cp addons/nuxeo-ai-image-quality-package/target/*package*.zip docker/ 2>/dev/null || echo "image-quality package not found"
+        cp addons/nuxeo-ai-aws-package/target/*package*.zip docker/ 2>/dev/null || echo "aws package not found"
+        cp addons/nuxeo-ai-gcp-package/target/*package*.zip docker/ 2>/dev/null || echo "gcp package not found"
+
+        echo "Docker context contents:"
+        ls -al docker || true
+
+    - name: Setup Docker Buildx
+      uses: docker/setup-buildx-action@v3
+      with:
+        driver: docker
+
+    - name: Login to Docker registry
+      uses: docker/login-action@v3
+      with:
+        registry: ${{ inputs.registry }}
+        username: ${{ inputs.docker-username }}
+        password: ${{ inputs.docker-password }}
+
+    - name: Build and push Docker image
+      uses: docker/build-push-action@v5
+      with:
+        context: ./docker
+        file: ./docker/Dockerfile
+        secrets: |
+          nuxeo-private=docker/nuxeo-private.repo
+        build-args: |
+          VERSION=${{ inputs.version }}
+          SCM_REF=${{ github.sha }}
+          BUILD_TAG=${{ github.run_id }}
+          PACKAGES_AUTH_USER=${{ inputs.docker-username }}
+          PACKAGES_AUTH_TOKEN=${{ inputs.docker-password }}
+        tags: |
+          ${{ inputs.registry }}/${{ inputs.repository }}:${{ inputs.version }}-${{ github.run_id }}
```

This notation will cause issue with semver sorting. Use instead:

```suggestion
          ${{ inputs.registry }}/${{ inputs.repository }}:${{ inputs.version }}.${{ github.run_id }}
```